src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -530,21 +531,28 @@ public async Task TestPartialProjectSync_GetSolutionFirst()
 
             solution = project2.Solution;
 
-            var assetProvider = await GetAssetProviderAsync(workspace, remoteWorkspace, solution);
+            var map = new Dictionary<Checksum, object>();
+            var assetProvider = new AssetProvider(
+                Checksum.Create(ImmutableArray.CreateRange(Guid.NewGuid().ToByteArray())), new SolutionAssetCache(), new SimpleAssetSource(workspace.Services.GetService<ISerializerService>(), map), remoteWorkspace.Services.GetService<ISerializerService>());
+
+            // Do the initial full sync
+            await solution.AppendAssetMapAsync(map, CancellationToken.None);
 
             var solutionChecksum = await solution.CompilationState.GetChecksumAsync(CancellationToken.None);
-            var syncedFullSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, solutionChecksum, updatePrimaryBranch: false, workspaceVersion: -1, CancellationToken.None);
+            var syncedFullSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, solutionChecksum, updatePrimaryBranch: true, workspaceVersion: solution.WorkspaceVersion, CancellationToken.None);
```

Comment:
note: when doing the initial full sync, we need to say `updatePrimaryBranch: true` and give a real version.  If we pass `false` or if we pass `-1` as the workspace version, then the remote side doesn't actually make this the 'CurrentSolution' of the workspace. 

So we can't actually validate that the cone forking is working (i.e. syncing the code, but keeping around the other untouched projects).